### PR TITLE
[ticket/16913] Add Search Index Progress Bar with Stats [master]

### PR DESCRIPTION
**Pull request by marc1706**
Based on #6410 for master, replaces #6411

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [ ] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-16913

### CODE REVIEW

File: phpBB/includes/acp/acp_search.php [hunk 1]
```diff
@@ -428,42 +456,73 @@ private function index_action(string $id, string $mode, string $action): void
...
 	 */
-	private function close_popup_js(): string
+	protected function get_post_index_progress(int $post_counter): array
 	{
-		return "<script type=\"text/javascript\">\n" .
-			"// <![CDATA[\n" .
-			"	close_waitscreen = 1;\n" .
-			"// ]]>\n" .
-			"</script>\n";
+		global $db;
```
Comment (rubencm) [inline phpBB/includes/acp/acp_search.php:502]: this->db

File: phpBB/includes/acp/acp_search.php [hunk 2]
```diff
@@ -392,6 +392,13 @@ private function index_action(string $id, string $mode, string $action): void
 			}
 		}
 
+		// Start displaying progress on first submit
```
Comment (rubencm) [inline phpBB/includes/acp/acp_search.php:395]: The best way to know if it's the first time you've entered here, I think it's putting it a few lines above, and you don't have to look at the post variables

File: phpBB/includes/acp/acp_search.php [hunk 3]
```diff
@@ -409,15 +416,36 @@ private function index_action(string $id, string $mode, string $action): void
...
+					$status['post_counter']
+				);
+				$message_redirect_rate = $this->language->lang(
+					($action === 'create') ? 'SEARCH_INDEX_CREATE_REDIRECT_RATE' : 'SEARCH_INDEX_DELETE_REDIRECT_RATE',
+					$status['rows_per_second']
+				);
+
+				$this->template->assign_vars([
+					'INDEXING_TITLE'		=> $this->language->lang($message_progress),
+					'INDEXING_EXPLAIN'		=> $this->language->lang($message_progress_explain),
```
Comment (GanstaZ) [inline phpBB/includes/acp/acp_search.php:433]: If my vision is not playing tricks on me, then here we see re-assigning language vars.. Previously assigned on lines 419-420.

Comment (marc1706) [reply phpBB/includes/acp/acp_search.php]: You're right, resolved now.

**Pull request merged**